packages/flutter/lib/src/widgets/list_wheel_scroll_view.dart

```diff
@@ -0,0 +1,713 @@
+       ),
+       assert(itemExtent != null && itemExtent > 0.0),
+       assert(clipToSize != null),
+       assert(renderChildrenOutsideViewport != null),
+       assert(
+         !renderChildrenOutsideViewport || !clipToSize,
+         'Cannot renderChildrenOutsideViewport and clipToSize since children'
+         'rendered outside will be clipped anyway, leading to gratuitous waste'
+       ),
+       super(key: key);
+
+  /// An object that can be used to control the position to which this scroll
+  /// view is scrolled.
+  ///
+  /// A [ScrollController] serves several purposes. It can be used to control
+  /// the initial scroll position (see [ScrollController.initialScrollOffset]).
+  /// It can be used to control whether the scroll view should automatically
+  /// save and restore its scroll position in the [PageStorage] (see
+  /// [ScrollController.keepScrollOffset]). It can be used to read the current
+  /// scroll position (see [ScrollController.offset]), or change it (see
+  /// [ScrollController.animateTo]).
+  final ScrollController controller;
+
+  /// How the scroll view should respond to user input.
+  ///
+  /// For example, determines how the scroll view continues to animate after the
+  /// user stops dragging the scroll view.
+  ///
+  /// Defaults to matching platform conventions.
+  final ScrollPhysics physics;
+
+  /// {@macro flutter.widgets.wheelList.diameterRatio}
+  ///
+  /// Defaults to an arbitrary but aesthetically sane number of 2.0.
+  ///
+  /// Must not be null and must be positive.
+  final double diameterRatio;
+
+  /// Perspective of the cylindrical projection.
+  ///
+  /// A number between 0.01 and 0 where 0 means looking at the cylinder from
+  /// infinitely far with an infinitely small field of view and 1 means looking
+  /// at the cylinder from infinitely close with an infinitely large field of
+  /// view (and also un-renderable).
+  ///
+  /// Defaults to an arbitrary but aesthetically sane number of 0.003. A larger
+  /// number brings the vanishing point closer and a smaller number pushes the
+  /// vanishing point further.
+  ///
+  /// Must not be null.
+  final double perspective;
+
+  /// Size of all children in the main axis. Must not be null and must be
+  /// positive.
+  final double itemExtent;
+
+  /// Whether to clip painted children to the inside of this scroll view.
+  ///
+  /// Defaults to [true]. Must not be null.
+  ///
+  /// If this is false and [renderChildrenOutsideViewport] is true, the
+  /// first and last children may extend outside.
+  final bool clipToSize;
+
+  /// Whether to paint children inside the viewport only.
+  ///
+  /// If false, every child will be painted. However the [Scrollable] is still
+  /// the size of the viewport and detects gestures inside only.
+  ///
+  /// Defaults to [false]. Must not be null. Cannot be true if [clipToSize]
+  /// is also true since children outside the viewport will be clipped, leading
+  /// to gratuitous waste.
```

Comment:
rather than "leading to gratuitous waste" I would just say "and therefore could not be painted outside the viewport".

Reply:
Done